src/WebApi/UnitTests/Tests/Application/UseCase/QuestionSetTests/DeleteQuestionSetUseCaseTest.cs

```diff
@@ -0,0 +1,83 @@
+﻿using Application.Repositories;
+using Application.UseCases.QuestionSet.DeleteQuestionSet;
+using Domain.Models;
+using Moq;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace UnitTests.Tests.Application.UseCase.QuestionSetTests
+{
+    public class DeleteQuestionSetUseCaseTest
+    {
+        private static DeleteQuestionSetInput Input => new DeleteQuestionSetInput()
+        {
+            Id = 1
+        };
+
+        private static QuestionSetModel Model => new QuestionSetModel()
+        {
+            Id = 1,
+            Title = "test",
+            Description = "my list",
+        };
+
+
+        [Fact]
+        public async Task Execute_RepositoryNull_Throws()
+        {
+            var useCase = new DeleteQuestionSetUseCase(null);
+
+            var execute = async () => await useCase.Execute(Input);
+
+            await Assert.ThrowsAsync<NullReferenceException>(execute);
+        }
+
+        [Fact]
+        public async Task Execute_PassEmpty_Throws()
+        {
+            var useCase = new DeleteQuestionSetUseCase(It.IsAny<IQuestionSetRepository>());
+
+            var execute = async () => await useCase.Execute(It.IsAny<DeleteQuestionSetInput>());
+
+            await Assert.ThrowsAsync<NullReferenceException>(execute);
+        }
+
+        [Fact]
+        public async Task Execute_PassValidInput_CallsNoContent()
+        {
+            var repositoryMock = new Mock<IQuestionSetRepository>();
+            repositoryMock.Setup(x => x.GetById(1).Result).Returns(Model);
+            repositoryMock.Setup(x => x.Delete(1).Result).Returns(true);
+            //repositoryMock.Setup(x => x.RemoveQuestionsFromList(1, It.IsAny<IEnumerable<int>>()).Result).Returns(true);
```

Comment:
To keep the solution as clean as possible we should eliminate all dead code.  Here is my suggestion :) 
```suggestion
```